src/custom/state/claim/hooks/index.ts

```diff
@@ -279,6 +279,17 @@ export function useUserClaims(account: Account, optionalChainId?: SupportedChain
   return { claims: claimKey ? claimInfo[claimKey] : null, isLoading }
 }
 
+let FETCH_DEPLOYMENT_TIMESTAMP_PROMISE: Promise<number> | null = null
+function fetchDeploymentTimestamp(vCowContract: VCowType) {
+  return (
+    FETCH_DEPLOYMENT_TIMESTAMP_PROMISE ??
```

Comment:
I personally think here is better to use just a `IF` because then the code is easier to read. Instead  of a single big statement, we break logic a bit in different lines

if (!FETCH_DEPLOYMENT_TIMESTAMP_PROMISE) {
    FETCH_DEPLOYMENT_TIMESTAMP_PROMISE = ...
}

return FETCH_DEPLOYMENT_TIMESTAMP_PROMISE

Reply:
Updated